contracts-available/FlightDelayStaking.sol

```diff
@@ -32,230 +48,245 @@ contract FlightDelayRiskPool is Ownable {
-     * Currently we use a fixed factor e.g. 1.0
-     *
-     * @param _stableStake
-     * @returns _requiredStake Amount of DIP tokens required.
-     *
+     * @dev get Stake of the address
+     * @param staker is the address of the staker
      */
-    function getCapacity() public view returns (uint256 _capacity) {
-        if (targetStake.stable < lockedStake.stable) {
-            return 0;
-        }
-        return targetStake.stable - lockedStake.stable;
+    function getStake(address staker)
+        public
+        view
+        returns (Stake memory _stake)
+    {
+        return stakeBalances[staker];
     }
 
     /**
-     * Calculate the required DIP tokens to stake the requested amount of stable asset
-     * Currently we use a fixed factor e.g. 1.0
-     *
-     * @param _stableStake
-     * @returns _requiredStake Amount of DIP tokens required.
-     *
+     * @dev get locked stake for staker
+     * @param _staker staker address
+     * @return _amount locked amount
      */
-    function lockCapacity(uint256 _capacity) public {
-        // add _capacity to lockedStake
+
+    function getLockedStakeFor(address _staker)
+        public
+        view
+        returns (uint256 _amount)
+    {
+        if (currentStake == 0) return 0;
+
+        uint256 stakedDai = stakeBalances[_staker].stable;
+
+        uint256 locked =
+            stakedDai
+                .div(currentStake)
+                .mul(totalLockedStake())
+                .mul(collatFactor)
+                .div(100);
+
+        return locked;
     }
 
     /**
-     * Calculate the required DIP tokens to stake the requested amount of stable asset
-     * Currently we use a fixed factor e.g. 1.0
-     *
-     * @param _stableStake
-     * @returns _requiredStake Amount of DIP tokens required.
-     *
+     * @dev get unlocked stake for staker
+     * @param _staker staker address
+     * @return _amount unlocked amount
      */
-    function unlockCapacity(uint256 _capacity) public {
-        // subtract _capacity from lockedStake
-        // process unstakeRequests
+
+    function getUnlockedStakeFor(address _staker)
+        public
+        view
+        returns (uint256 _amount)
+    {
+        Stake memory _stake = stakeBalances[_staker];
+
+        return _stake.stable - getLockedStakeFor(_staker);
     }
 
     /**
-     * Calculate the required DIP tokens to stake the requested amount of stable asset
-     * Currently we use a fixed factor e.g. 1.0
-     *
-     * @param _stableStake
-     * @returns _requiredStake Amount of DIP tokens required.
-     *
+     * @dev get maximum exposure
+     * @return _capacity is maximum exposure
      */
-    function addStake(stake _s1, stake _s2) public pure returns (_stake _result) {
-        _result.dip = _s1.dip + _s2.dip;
-        _result.stable = _s1.stable + _s2.stable;
+    function maximumCapacity() public view returns (uint256 _capacity) {
```

Comment:
we should have an additional "getMaximumAvailableCapacity" function 
maximumAvailableCapacity = maximumCapacity - usedCapacity